packages/web/components/one-click-trading/one-click-trading-settings.tsx

```diff
@@ -170,11 +201,22 @@ export const OneClickTradingSettings = ({
   ) => {
     setTransaction1CTParamsProp((prevParams) => {
       const nextParams = runIfFn(newParamsOrFn, prevParams);
+
+      if (!initialTransaction1CTParams || !nextParams) {
+        console.error("Transaction params are undefined");
+        return prevParams;
+      }
+
       setChanges(
-        compare1CTTransactionParams({
-          prevParams: initialTransaction1CTParams!,
-          nextParams: nextParams!,
-        })
+        Array.from(
+          new Set([
+            ...changes,
+            ...compare1CTTransactionParams({
+              prevParams: initialTransaction1CTParams!,
+              nextParams: nextParams!,
+            }),
+          ])
```

Comment:
What situations are we preventing here? Could you add a comment explaining why we need to spread the `changes` state?

Reply:
ok, this was the most difficult request so far :)

Reply:
Context: I have a habit of leaving details and explanations in commit messages for the given change so it's immediately obvious clicking on a line why a change was made.

I would like to understand if this is a good practice on this project or should I prefer inline comments for similar explanations?